sql/sql_base.cc

```diff
@@ -4522,6 +4523,7 @@ bool open_tables(THD *thd, const DDL_options_st &options,
               goto error;
 
             error= FALSE;
+            std::this_thread::yield();
             goto restart;
```

Comment:
This is duplicating the section of code 63 lines earlier. In fact, a wider section of the code block starting with `if (unlikely(error))` is being duplicated. I think that a follow-up fix would be useful where we’d do the following:
```c++
  if (unlikely(error))
  {
    if (!(error= open_tables_handle_error(error, …)))
      goto restart;
    goto error;
  }
```
and there would be a new function `ATTRIBUTE_COLD static bool open_tables_error_handler(bool, …)` that would include the duplicated code, including the call to `std::this_thread::yield()`.

Reply:
Right. There was a change in 10.11 though, that makes it different by one line at least. I wouldn't dare doing it in GA versions though.